namespace.c

```diff
@@ -464,7 +464,7 @@ const rb_data_type_t rb_namespace_data_type = {
         namespace_entry_memsize,
         rb_namespace_gc_update_references,
     },
-    0, 0, RUBY_TYPED_WB_PROTECTED
+    0, 0, // TODO: enabling RUBY_TYPED_WB_PROTECTED would be good, but write barriers should be inserted.
```

Comment:
Don't we need `RUBY_TYPED_FREE_IMMEDIATELY` instead?

Reply:
It's not instead we likely want both. `RUBY_TYPED_FREE_IMMEDIATELY` can be added without modifying anything, it basically means: "my free function doesn't do IO or things that could block". So yes, you can add it.